### PR TITLE
nixos/frr: adapt to frr-9

### LINKED ISSUE #274286: nixos/frr: staticd does not accept -f anymore in frr-9
### Describe the bug
using frr-9.0.1 from nixos-23.11 results in failure of staticd.service:

`/nix/store/blgvhysfkq43jsv5p0yzmqn4j0q6aqd5-frr-9.0.1/libexec/frr/staticd: invalid option -- 'f'`

### Steps To Reproduce
services.frr.static.enable = true
systemctl status staticd

### Expected behavior
staticd used to work with `-f configfile` option

### Additional context
see
- https://github.com/FRRouting/frr/issues/14040
- https://docs.frrouting.org/en/latest/mgmtd.html
- https://github.com/FRRouting/frr/blob/master/doc/developer/mgmtd-dev.rst

Quick fix see below.

maybe @mweinelt / @thillux want to insert themselves kindly into maintainers :-)

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.67, NixOS, 23.11 (Tapir), 23.11.20231212.cf28ee2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(fd): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/dg2g5qwvs36dhfqj9khx4sfv0klwl9f0-source`

### PR DESCRIPTION
**Pull request by woffs**
- fix #274286
- remove `-f configfile` from ExecStart
- use /etc/frr/${service}d.conf
- enable mgmtd when staticd is enabled
- don't frr-reload.py mgmtd
- remove obsolete lib.mdDoc

## Description of changes
In frr-9 staticd no longer accepts `-f configfile` option, but uses mgmtd for configuration management. Therefore we remove `-f configfile` from all frr services and use the default config location `/etc/frr/${service}d.conf` (NixOS used this without `d` before) and enable mgmtd automatically, if staticd is enabled. We have to follow upstream release notes carefully if they add more daemons to the new mgmtd method (or we start mgmtd, like zebra, if any service is enabled?).

I had no leasure to read and understand mgmtd documentation thoroughly yet, sorry. I'm not sure how to handle mgmtd reload, I tried to disable reloading mgmtd for now, because frr-reload denies reloading mgmtd.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Review by thillux — APPROVED** [review]
LGTM

As NixOS currently generates frr daemon configs into store paths, which are read-only, we can keep separate daemon configs also for frr 10.0. A unified frr.conf would currently only be necessary, if writing to this file should also be supported via e.g. vtysh.

**Comment by mweinelt** [comment]
Probably needs to be rebased. `lib.mdDoc` was removed treewide a few days ago.

**Comment by woffs** [comment]
> Probably needs to be rebased. `lib.mdDoc` was removed treewide a few days ago.

done

**Comment by woffs** [comment]
@GrahamcOfBorg test frr

**Comment by thillux** [comment]
Regarding mgmtd reload: mgmtd keeps track of different config versions for other daemons. It cannot be reloaded with a different config on its own. I think its ok to block/ignore this operation. frr-reload.py also does not contain mgmtd in the list of allowed daemons to reload (https://github.com/FRRouting/frr/blob/master/tools/frr-reload.py#L1999). I guess from semantics side that's correct.

**Comment by thillux** [comment]
@woffs What's your opinion regarding mgmtd?

**Comment by woffs** [comment]
> @woffs What's your opinion regarding mgmtd?

If it cannot be reloaded and does not need to be reloaded, we should not reload it :wink:

File: nixos/modules/services/networking/frr.nix [hunk 1]
```diff
@@ -137,6 +136,19 @@ in
             '';
           };
         };
+        mgmt = (serviceOptions "mgmt") // {
+          enable = mkOption {
+            type = types.bool;
+            default = isEnabled "static";
```
Comment (mweinelt) [inline nixos/modules/services/networking/frr.nix:142]: Please add a `defaultText` that explains this relationship.

Comment (woffs) [reply nixos/modules/services/networking/frr.nix:142]: I see, doing
`defaultText = lib.literalExpression "config.services.frr.static.enable";`

Comment (mweinelt) [reply nixos/modules/services/networking/frr.nix:142]: Yeah, that would work.

> "this is when staticd is enabled" in description is not sufficient?

Expressions as the default don't render well on search.nixos.org iirc.

**Pull request merged**